### PR TITLE
Extend path-deps extension to handle explicit path refreshes

### PR DESCRIPTION
**Pull request by jvosloo**
This is to implement #286.

**TLDR**: Went with `PathDeps.refresh("/path/to/refresh");` in the end.
Keeping the rest here for historical purposes.

-----

By catering for a new custom event called `htmx:refreshPath`, we should be able to trigger the refresh of a specific path, e.g. 

- `htmx.trigger(document, "htmx:refreshPath", {path:"path/to/refresh"});`
- `htmx.trigger(document, "htmx:refreshPath", {path:"path/to/refresh/*"});`

I have not added any tests or documentation yet, happy to do so if this idea is deemed to be ok in principle.

**Note**: Since we're interested in triggering updates on all elements with the given path (or dependent paths), the element to trigger on is irrelevant, I just went with "document" in the example. Not sure if there's a better way

### CODE REVIEW

**Comment by jvosloo** [comment]
I had a blond moment and added and then promptly removed another event called `htmx:refreshElement`, but that's a little daft, because if you already know the element you want to refresh, you can just trigger the `path-deps` event on it directly, e.g. `htmx.trigger(htmx.find("#myEl"), "path-deps")`

**Review by 1cg — COMMENTED** [review]
Thank you for addressing this shortcoming compared with intercooler.

The logic looks good.  As far as the API goes, what about surfacing a global symbol, `PathDeps`, with a single method on it:

```js
  PathDeps.refresh("/some/path")
```

That way you don't have to go through htmx to get to the functionality?

**Comment by jvosloo** [comment]
> Thank you for addressing this shortcoming compared with intercooler.
> 
> The logic looks good. As far as the API goes, what about surfacing a global symbol, `PathDeps`, with a single method on it:
> 
> ```js
>   PathDeps.refresh("/some/path")
> ```
> 
> That way you don't have to go through htmx to get to the functionality?

Yours is a far better idea than what I had 🤦‍♂️.  I added the PathDeps api with the refresh method. Does it look ok?

**Comment by 1cg** [comment]
Looks reasonable to me!  Can you add tests for it?

**Comment by 1cg** [comment]
Pinging again: all this needs are some tests in `test/ext/path-deps.js` and I'm happy to pull

**Comment by jvosloo** [comment]
Sure, no problem. I'm away for a few days with the fam so I will get on it first thing next week 👍

**Comment by jvosloo** [comment]
@1cg I added a couple of tests and updated the documentation to reflect the addition of the API and the `refresh` method.
I couldn't really think of more useful tests to add, since the refresh method reuses the same logic that's already being extensively tested by the existing tests?

**Comment by 1cg** [comment]
Looks good, thank you for adding tests.  I don't understand this code though:

```js
    _root.PathDeps = {

        PathDeps.refresh: function(path) {
            refreshPath(path);
        }
    };
```

Why is the property named `PathDeps.refresh` rather than just `refresh`?

**Comment by 1cg** [comment]
Ah, looks like it was an accident.  I fixed it and all the tests pass...

**Comment by jvosloo** [comment]
> Ah, looks like it was an accident.  I fixed it and all the tests pass...

Yup, oversight. Don't know how I managed to leave that in there 🤦🏻‍♂️ Thanks for fixing :)
